[TF-11689] Update organization settings (#521)
hashicorp/terraform-docs-common: website/docs/cloud-docs/users-teams-organizations/organizations.mdx

#### General

> Organization owners can choose between _non-aggregated_ (default) and _aggregated_ status checks. This setting determines whether detailed information and links are accessed directly from the VCS provider or Terraform Cloud.

Reviewer: This now disagrees with the API docs about what the default is

```suggestion
Organization owners can choose between _aggregated_ (default) and _non-aggregated_ status checks. This setting determines whether detailed information and links are accessed directly from the VCS provider or Terraform Cloud.
```

This setting also determines the number of status checks directly sent to the VCS Provider in response to actions such as pull or merge requests.

> ##### Non-aggregated Status Checks

Author: Thanks for catching this.

![Screenshot: Organization Non-aggregated status checks](/img/docs/organization-vcs-general-non-aggregated-status-checks.png)

> The default behavior of VCS providers is that a provider receives a status for each triggered workspace and related run stage in response to a VCS event. For example, a VCS push triggers checks for each related workspace's run stages, including the plan operation, policy checks, cost estimation, run tasks, and more.

Author: Done, thank you 🙏.

You can disable this behavior if it creates too many status checks to your VCS provider. You may want to do this if you have a large number of workspaces sharing one VCS repository.

##### Aggregated Status Checks

![Screenshot: Organization Aggregated status checks](/img/docs/organization-vcs-general-aggregated-status-checks.png)

Aggregated status checks offer a streamlined experience if you have a single repository containing configuration for many workspaces (a.k.a., a monorepo).

When aggregated status checks are enabled, Terraform Cloud sends one VCS status check for all runs triggered by a VCS event. If multiple workspaces are tied to a shared repository, Terraform Cloud aggregates the status checks for these workspaces into one summary. This summary is unique to the workspace's organization and VCS client connection.

> You can access additional information about an aggregated status check in Terraform Cloud by clicking the **Details** link that _X_ provides. This link navigates you to a Terraform Cloud page that offers the consolidated status check results across multiple workspaces, highlighting details such as resource changes and issues that require attention.

Reviewer: What is the _X_ here?

Should this be

```suggestion
You can access additional information about an aggregated status check in Terraform Cloud by clicking the **Details** link that the status check provides. This link navigates you to a Terraform Cloud page that offers the consolidated status check results across multiple workspaces, highlighting details such as resource changes and issues that require attention.
```

#### General

> 

Reviewer 2: I do think we should briefly describe what a status check is, what it does, and why TFC uses it in this section.

> Organization owners can choose between Non-Aggregated (Default) and Aggregated status checks. This setting determines whether detailed information and links are accessed directly from the VCS Provider or Terraform Cloud.

Reviewer 2: 1. Where can people find this setting? Like what is the label or name for the UI element that folks should look for (so folks can ctrl+f it)? 
2. what is the topic of the detailed information and links? The status check results? Also, the way this is currently worded it sounds like flipping this setting changes if the VCS provider or TFC accesses the link? That could be true, and if so we should explicitly

> Organization owners can choose between Non-Aggregated (Default) and Aggregated status checks. This setting determines whether detailed information and links are accessed directly from the VCS Provider or Terraform Cloud.

Author: 1. The setting will be a toggle between, `Aggregated status checks` and `Non-aggregated status checks`
2. Somewhat explain what the detailed link to Terraform Cloud would include in the description below
3. Its the latter where you as the user can access and visibly see the information. Existing behavior is you can view all status checks across all impacted workspaces, however with the new feature on, you the user will have to click the link to a page in Terraform Cloud to see runs statuses spanning multiple workspaces

> It also determines the quantity of status checks directly sent to the VCS Provider in response to actions such as Pull/Merge requests.

Reviewer 2: ```suggestion
This setting also determines the number of status checks directly sent to the VCS Provider in response to actions such as pull or merge requests.
```

> ##### Aggregated Status Checks

Reviewer 2: Can you add an example here in the beginning or at the end? We state who this is for, but not _why_ they should do it. What does this look like in someone's setup? What is the advantage they gain by doing this? What are the tradeoffs between this and the default behavior?

> Aggregated status checks offer a streamlined experience to customers with "monorepo" repositories, containing configuration for a large number of workspaces.

Reviewer 2: In the docs we like to talk directly to the user, so we avoid using terms like "customer" in favor of directly using "you":
```suggestion
Aggregated status checks offer a streamlined experience if you have a single repository containing configuration for many workspaces (a.k.a., a monorepo).
```

> When enabled, Terraform Cloud will send one VCS status check for all runs triggered by a VCS event. If multiple workspaces are tied to a shared repository, status checks for these workspaces will be aggregated into one summary. This summary is unique to the workspace's organization and VCS client connection.

Reviewer 2: ```suggestion
When aggregated status checks are enabled, Terraform Cloud sends one VCS status check for all runs triggered by a VCS event. If multiple workspaces are tied to a shared repository, Terraform Cloud aggregates the status checks for these workspaces into one summary. This summary is unique to the workspace's organization and VCS client connection.
```

> The aggregated status check will allow users to access additional information in Terraform Cloud by accessing the provided "details" link. This link navigates users to a page in Terraform Cloud that offers a consolidated presentation across multiple Workspaces, highlighting details such as resource changes and possible issues that require user's attention.

Reviewer 2: Where do users see the link for "details"? In their VCS workflow? If they actually have something labled "details" then keep the bold below:
```suggestion
You can access additional information about an aggregated status check in Terraform Cloud by clicking the **Details** link that _X_ provides. This link navigates you to a Terraform Cloud page that offers the consolidated status check results across multiple workspaces, highlighting details such as resource changes and issues that require attention.
```

> ##### Non-aggregated Status Checks

Reviewer 2: I suggest the default behavior should come first; that way folks understand what this looks like without doing anything. Then with that content in mind, they can decide to turn aggregated status checks on or off

> Default behavior where the VCS provider receives a status for each triggered Workspace and related Run stages (e.g. Plan, Policy Checks, Cost Estimation, Run Tasks, etc.) in response to a VCS event.

Reviewer 2: I tried spreading this out a bit for clarity. Please feel free to tweak that example if it's not accurate 😅 
```suggestion
The default behavior of VCS providers is that a provider receives a status for each triggered workspace and related run stage in response to a VCS event. For example, a VCS push triggers checks for each related workspace's run stages, including the plan operation, policy checks, cost estimation, run tasks, and more. 
```

###### Send passing commit statuses

> -> **Note:** Organization owners can only enable **Send passing commit statuses** option if **Aggregated status checks** is disabled.

Reviewer 2: Do we also need a corresponding note above?  So folks know they can't enable aggregated status checks unless they have **Send passing commit statuses** disabled?
```suggestion
-> **Note:** Organization owners can only enable the **Send passing commit statuses** setting if the **Aggregated status checks** setting is disabled.
```

Author: I don't think we do since the settings page experience makes this option visible only if it's disabled

> 

Reviewer 2: If we want this feature to be _discoverable_, I do think we should make a new page for this content. This page is so _long_, and this content is quite buried within it😅 

However, if this is a quick fix before the TFC GA and we'll follow up with a new page later, that makes sense to me!

Author: Agreed, this is just for TFC GA

> #### General

Reviewer 2: Sorry to nitpick, but I don't think we need the "General" header here,  I know it's in the UI, but people aren't going to be searching for it and the stacked headers look silly as is:
```suggestion
```
I think it looks better without the extra header:
![Screenshot 2024-01-03 at 11 18 43 AM](https://github.com/hashicorp/terraform-docs-common/assets/32436232/62b46ac2-128a-4a99-ac03-613b8d26502e)

###### Aggregated Status Checks

Aggregated status checks offer a streamlined experience if you have a single repository containing configuration for many workspaces (a.k.a., a monorepo).

When aggregated status checks are enabled, Terraform Cloud sends one VCS status check for all runs triggered by a VCS event. If multiple workspaces are tied to a shared repository, Terraform Cloud aggregates the status checks for these workspaces into one summary. This summary is unique to the workspace's organization and VCS client connection.

You can access additional information about an aggregated status check in Terraform Cloud by clicking the **Details** link that the status check provides. This link navigates you to a Terraform Cloud page that offers the consolidated status check results across multiple workspaces, highlighting details such as resource changes and issues that require attention.

![Screenshot: Organization Aggregated status checks](/img/docs/organization-vcs-general-aggregated-status-checks.png)

###### Non-aggregated Status Checks

> The default behavior of VCS providers is that a provider receives a status for each triggered workspace and related run stage in response to a VCS event. For example, a VCS push triggers checks for each related workspace's run stages, including the plan operation, policy checks, cost estimation, run tasks, and more.

Reviewer 3: Maybe change the beginning of the paragraph to `The alternative behavior` or something else?

Author: thanks for the catch. Went back and forth a few times on which is the "default"
